pkg/handler/handler.go

```diff
@@ -118,8 +118,9 @@ func WebhookHandler(w http.ResponseWriter, r *http.Request, reconciler gitopscon
 }
 
 func repoURLAndRefMatch(instance *gitopsv1alpha1.GitOpsConfig, event *github.PushEvent) bool {
-	return (strings.Contains(instance.Spec.TemplateSource.URI, *event.Repo.FullName) &&
-		instance.Spec.TemplateSource.Ref == strings.TrimPrefix(*event.Ref, "refs/heads/")) ||
+	return event.Repo != nil && event.Repo.FullName != nil && event.Ref != nil &&
+		(strings.Contains(instance.Spec.TemplateSource.URI, *event.Repo.FullName) &&
```

Comment:
I think the `||` block needs to be in an extra pair of parentheses...

NIL_CHECKS && ((CONTAINS_CHECK && REF_CHECK) || (CONTAINS_CHECK && REF_CHECK))

otherwise I think the precedence rules will group all `&&`s together

Reply:
Fixed